tests/sql_parse_tests.py

```diff
@@ -16,90 +16,102 @@
 # under the License.
 import unittest
 
-from superset import sql_parse
+from superset.sql_parse import ParsedQuery, Table
 
 
 class SupersetTestCase(unittest.TestCase):
     def extract_tables(self, query):
-        sq = sql_parse.ParsedQuery(query)
-        return sq.tables
+        return ParsedQuery(query).tables
+
+    def test_table(self):
+        self.assertEqual(str(Table("tbname")), "tbname")
+        self.assertEqual(str(Table("tbname", "schemaname")), "schemaname.tbname")
+
+        self.assertEqual(
+            str(Table("tbname", "schemaname", "catalogname")),
+            "catalogname.schemaname.tbname",
+        )
+
+        self.assertEqual(
+            str(Table("tb.name", "schema/name", "catalog\name")),
+            "catalog%0Aame.schema%2Fname.tb%2Ename",
+        )
```

Comment:
whoops, it's picking `\name` as linechange +`ame` 😄

Reply:
Yip. Does it mater though?

Reply:
Yeah, for context, this quoting is what's used to make the special characters usable in SqlAlchemy URIs. Not sure what the convention is for fully qualified names outside SqlAlchemy scope, but as long as we have a convention that works both ways we can always encode into fully qualified format/decode into catalog.schema.table as needed.